rotors_driver/src/rotors_driver_node.py

```diff
@@ -0,0 +1,356 @@
+										 header = Header(frame_id = self.frame_id))
+		self.rqt_velocity_body_publisher.publish(twist)
+
+
+		self.rqt_cam_frontal_publisher.publish(self.frontal_image)
+		self.rqt_cam_ventral_publisher.publish(self.ventral_image)
+
+
+	def cam_frontal(self, msg):
+		self.frontal_image = msg
+		rospy.logdebug('Frontal image updated')
+
+	def cam_ventral(self, msg):
+		self.ventral_image = msg
+		rospy.logdebug('Ventral image updated')
+
+	def get_pose(self,msg):
+		self.current_state = msg
+		
+		self.current_x=self.current_state.pose.pose.position.x
+		self.current_y=self.current_state.pose.pose.position.y
+		self.current_z=self.current_state.pose.pose.position.z
+		self.current_vx=self.current_state.twist.twist.linear.x
+		self.current_vy=self.current_state.twist.twist.linear.y
+		self.current_vz=self.current_state.twist.twist.linear.z
+		self.current_ang_vx=self.current_state.twist.twist.angular.x
+		self.current_ang_vy=self.current_state.twist.twist.angular.y
+		self.current_ang_vz=self.current_state.twist.twist.angular.z
+		self.quaternion = [self.current_state.pose.pose.orientation.x, self.current_state.pose.pose.orientation.y,
+					self.current_state.pose.pose.orientation.z, self.current_state.pose.pose.orientation.w
+				]
+		self.current_roll, self.current_pitch, self.current_yaw = tf.transformations.euler_from_quaternion(self.quaternion)
+		self.current_yaw_rate = self.current_state.twist.twist.angular.z
+
+		self.frame_id = self.current_state.header.frame_id
+
+		if self.current_z<0.1:
+			self.__is_flying = False
+		else:
+			self.__is_flying = True
+
+
+	def rotors_takeoff_land(self,req):
+		quaternion = tf.transformations.quaternion_from_euler(0, 0, 0.0)
+		traj = MultiDOFJointTrajectory()
+		
+		header = Header()
+		header.stamp = rospy.Time()
+		header.frame_id = 'frame'
+		traj.joint_names.append('base_link')
+		traj.header=header
+		
+		if req:
+			transforms =Transform(translation=Point(0.0, 0.0, 1.0), rotation=Quaternion(quaternion[0],quaternion[1],quaternion[2],quaternion[3]))
+			velocities =Twist()
+			accelerations=Twist()
+		else:
+			transforms =Transform(translation=Point(self.current_x, self.current_y, 0.0), rotation=Quaternion(quaternion[0],quaternion[1],quaternion[2],quaternion[3]))
+			velocities =Twist(linear = Vector3(x=0.0, y=0.0, z=0.0), angular=Vector3(x=0.0, y=0.0, z=0.0) )
+			accelerations=Twist(linear = Vector3(x=0.0, y=0.0, z=0.0), angular=Vector3(x=0.0, y=0.0, z=0.0) )
+
+		
+		point = MultiDOFJointTrajectoryPoint([transforms],[velocities],[accelerations],rospy.Time(2))
+
+		traj.points.append(point)
+		
+
+		self.firefly_command_publisher.publish(traj)
+		time.sleep(1) #change the time if altitude val doesn't change
+		if req and 0.8<self.current_z<1.2:
+			return True, 0
+		elif not req and 0.0<=self.current_z<0.3:
+			rospy.loginfo("Firefly Disarming")
+			self.__is_armed = False
```

Comment:
Not sure if disarming when I might be flying is a good idea. If I'm landing at 0.2m height I'll be disarmed but flying, right?

Reply:
solved in latest changes..these were some unresolved contradictions and I've solved with little changes